### ISSUE TITLE
Kokkos Develop causes Segmentation Fault in OpenMP Dispatch Test for PGI 17.7.0

### ISSUE THREAD

**Issue by nmhamster** [kokkos/kokkos#1181]
```
../generate_makefile.bash --with-openmp --arch=HSW --compiler=`which pgc++`
```

Then run tests results in:

```
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
[==========] Running 89 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 89 tests from openmp
[ RUN      ] openmp.init
[       OK ] openmp.init (0 ms)
[ RUN      ] openmp.dispatch
Segmentation fault
```

**Comment by nmhamster** [comment]
Backtrace from segmentation fault:

```
Program received signal SIGSEGV, Segmentation fault.
_IO_vfprintf_internal (s=0xfffffffff68a1880, format=0x7ffff703c288 "Requested invalid number of cpus: %d\n",
    ap=0x7fffffffcf60) at vfprintf.c:1283
1283	  ORIENT;
Missing separate debuginfos, use: debuginfo-install numactl-2.0.7-8.el6.x86_64
(gdb) bt
#0  _IO_vfprintf_internal (s=0xfffffffff68a1880, format=0x7ffff703c288 "Requested invalid number of cpus: %d\n",
    ap=0x7fffffffcf60) at vfprintf.c:1283
#1  0x00007ffff65620f8 in __fprintf (stream=<value optimized out>, format=<value optimized out>) at fprintf.c:33
#2  0x00007ffff7039b91 in _mp_create_team ()
   from /home/projects/x86-64/pgi/2017/17.7.0/linux86-64/17.7/lib/libpgmp.so
#3  0x00007ffff7039d68 in _mp_cpenter () from /home/projects/x86-64/pgi/2017/17.7.0/linux86-64/17.7/lib/libpgmp.so
#4  0x00007ffff702dc45 in _mp_penter64 () from /home/projects/x86-64/pgi/2017/17.7.0/linux86-64/17.7/lib/libpgmp.so
#5  0x00007fffffffd0f0 in ?? ()
#6  0x0000000000416dcb in testing::internal::UnitTestOptions::PatternMatchesString ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:1933
#7  0x00000000004745f4 in Kokkos::Impl::ParallelFor<void Test::test_dispatch<Kokkos::OpenMP>()::[lambda(int) (instance 1)], Kokkos::RangePolicy<<Kokkos::OpenMP>>, Kokkos::OpenMP>::execute() const(void) ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:132
#8  0x000000000047996b in void Kokkos::parallel_for<Kokkos::RangePolicy<<Kokkos::OpenMP>>, void Test::test_dispatch<Kokkos::OpenMP>()::[lambda(int) (instance 1)]>(void) ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/core/unit_test/openmp/TestOpenMP_Init.cpp:1571
#9  0x0000000000479715 in Test::test_dispatch<Kokkos::OpenMP> ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/core/unit_test/TestInit.hpp:64
#10 0x0000000000455b9b in Test::openmp_dispatch_Test::TestBody ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/core/unit_test/TestInit.hpp:72
#11 0x0000000000454785 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void> ()
#12 0x000000000045174a in testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void> ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:3595
#13 0x000000000040eebf in testing::TestInfo::Run ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:3634
#14 0x0000000000418f05 in testing::internal::UnitTestImpl::RunAllTests ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:3930
#15 0x0000000000454845 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> () at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:3562
#16 0x00000000004527fa in testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> () at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:3595
#17 0x000000000041016a in testing::UnitTest::Run ()
    at /ascldap/users/sdhammo/git/kokkos-github-repo/tpls/gtest/gtest/gtest-all.cc:5365
#18 0x000000000040bf1d in main ()
```

**Comment by nmhamster** [comment]
This is fixed in PGI 17.9.0 release.
